### PR TITLE
docs: reorganize client doc

### PR DESCRIPTION
**Pull request by jpetot**
I also update the python client. But my main work was to re-organize the client doc. 
see the picture to have an overview of the tree 😁
image

### CODE REVIEW

File: doc/source/how_to.rst [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+**NB**: This section is only available if the use of emails to login is
+activated in the global configuration. It is not the case by default.
+
+If you forgot your password or if you need to set one, first call the
+forgotten_password endpoint:
+
+.. raw:: html
+
+   <!-- It is python, just doing this so that test-integration does not run this code (need mail config to run)  -->
```
Comment (tas17) [inline doc/source/how_to.rst:16]: If it's rst, it's not tested anymore. so javascript -> python and remove the comment

**Review by tcrasset — COMMENTED** [review]
Very nice, this is a step forward to making the doc easier to read and use! Good job.

I just have some general comments:
- do we want to keep writing in `.md` or `.rst` (we can use either one, we have `pandoc` to switch between them). Personally, I find the former to have easier syntax to write:
	- no double backticks to have a variable,
	- no forcing to have the same number of characters for the underline of the title,
	- not a different character for different title size
	- no complicated syntax to have a code block
- you have split the `tutorial.md` notebook, which was fully tested with `make test-integration`
We'll have to make sure that any code we write in the `.rst` file is fully tested, perhaps we can use the same mechanism as we did for `tutorial.md`
We'll also have to make sure to use the tutorials in `notebooks/` as our integration test then, and run `make test-tutorial` instead of `make test-integration` everytime.

- Now that you've taken the time to split the different sections, shouldn't we reflect on what our file structure should be like? I feel like `how_to.rst` will become the new `tutorial.rst`, a big file with lots of unrelated parts.

File: doc/source/how_to.rst [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+   from avatars.client import ApiClient
+
+   client = ApiClient(base_url=os.environ.get("BASE_URL"))
+   client.forgotten_password("yourmail@mail.com")
+
+You’ll then receive an email containing a token. This token is only
+valid once, and expires after 24 hours. Use it to reset your password:
+
+.. code:: javascript
```
Comment (tas17) [inline doc/source/how_to.rst:28]: Same here

**Review by tas17 — COMMENTED** [review]
The new looking is way better IMO. But to me everything should be tested so that it remains up to date. It's not the case anymore.

**Review by tas17 — APPROVED** [review]

File: .github/workflows/lci.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        run: poetry install --no-interaction --no-root
+
+      - name: Install app
+        run: poetry install --no-interaction
+
+      - name: Test
+        run: 	poetry run pytest --doctest-modules avatars
+
+      - name: Build doc
+        run: make doc-fast
```
Comment (tcrasset) [inline .github/workflows/lci.yaml:58]:
```suggestion
      - name: Build latest doc version
        run: make doc-fast
```

File: doc/source/user_guide.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+```javascript
+from avatars.client import ApiClient
+
+client = ApiClient(base_url=os.environ.get("BASE_URL"))
+client.reset_password("yourmail@mail.com", "new_password", "new_password", "token-received-by-mail")
+```
+
+You’ll receive an email confirming your password was reset.
+
+## How to log in to the doc/source/user_guide.md server
```
Comment (tcrasset) [inline doc/source/user_guide.md:33]: What is the `doc/source/user_guide.md` server?

File: doc/source/user_guide.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+import io
+
+# Change this to your actual server endpoint, e.g. base_url="https://avatar.company.com"
+client = ApiClient(base_url=os.environ.get("BASE_URL"))
+client.authenticate(
+    username="username", password=os.environ.get("AVATAR_PASSWORD", "strong_password")
+)
+```
+
+## How to upload a dataframe
```
Comment (tcrasset) [inline doc/source/user_guide.md:52]:
```suggestion
## How to upload a data
```
You are wanting to expose how data is uploaded, both as a dataframe, and as a csv. So the title has to be generic

File: doc/source/user_guide.md [hunk 6]
```diff
@@ -0,0 +1,318 @@
...
+```python
+import pandas as pd
+
+df = pd.read_csv("fixtures/iris.csv")
+
+# ... do some modifications on the dataset
+
+dataset = client.pandas_integration.upload_dataframe(df)
+```
+as a ``.csv`` file
```
Comment (tcrasset) [inline doc/source/user_guide.md:65]:
```suggestion
### As a `.csv` file
```
double backticks are for `.rst`

File: doc/source/user_guide.md [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+```python
+filename = "fixtures/iris.csv"
+
+with open(filename, "r") as f:
+    dataset = client.datasets.create_dataset(request=f)
+```
+
+## How to launch an avatarization with metric
+
+You can lunch an avatarization with some simple privacy and signal metrics.
```
Comment (tcrasset) [inline doc/source/user_guide.md:76]:
```suggestion
You can launch an avatarization with some simple privacy and signal metrics.
```

ctrl + F 'lunch' at this point

File: doc/source/user_guide.md [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+    )
+)
+job = client.jobs.get_avatarization_job(job.id, timeout=10)
+print(job.status)
+print(job.result)
+```
+
+## How to launch privacy metrics
+
+You can launch a privacy metrics with two datasets, the original and the anonymized.
```
Comment (tcrasset) [inline doc/source/user_guide.md:115]:
```suggestion
You can launch a privacy metrics job with two datasets, the original and the anonymized.
```

File: doc/source/user_guide.md [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+    imputation=imputation,
+    exclude_categorical=exclude_parameters,
+)
+```
+
+## How to generate the report
+
+You can create an avatarization report. 
+
+You need to run privacy and signal metrics with the arguments ``persistance_job_id=job.id`` before running the report.
```
Comment (tcrasset) [inline doc/source/user_guide.md:216]:
```suggestion
```
This is not true anymore

File: doc/source/user_guide.md [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+        avatarization_job_create=AvatarizationJobCreate(
+            parameters=AvatarizationParameters(dataset_id=dataset.id, k=20),
+        ),
+        processors=[proportion_processor],
+        df=df,
+    )
+)
+```
+
+See [processors](processors.html) for more information about the processor.
```
Comment (tcrasset) [inline doc/source/user_guide.md:271]:
```suggestion
See [processors](processors.html) for more information about the processors. 
```

File: doc/source/user_guide.md [hunk 11]
```diff
@@ -0,0 +1,318 @@
...
+    )
+)
+```
+
+See [processors](processors.html) for more information about the processor. 
+See [this notebook](https://github.com/octopize/avatar-python/blob/main/notebooks/Tutorial4_Client_side_processors.ipynb) for an advanced usage of the pipeline.
+
+## How to download an avatar dataset 
+
+As a pandas dataframe.
```
Comment (tcrasset) [inline doc/source/user_guide.md:276]:
```suggestion
### As a pandas dataframe
```

File: doc/source/user_guide.md [hunk 12]
```diff
@@ -0,0 +1,318 @@
...
+
+```python
+result = job.result
+avatars_dataset_id = result.avatars_dataset.id
+
+avatar_df = client.pandas_integration.download_dataframe(avatars_dataset_id)
+print(avatar_df.head())
+```
+
+As a ``.csv`` file as string.
```
Comment (tcrasset) [inline doc/source/user_guide.md:289]:
```suggestion
### As a `.csv` formatted string
```

**Review by tcrasset — CHANGES_REQUESTED** [review]
Good, some fixes I'd like to see because they are out of date. Some are just nitpicks

**Pull request merged**